drizzlepac/hlautils/astrometric_utils.py

```diff
@@ -584,16 +586,44 @@ def extract_sources(img, dqmask=None, fwhm=3.0, kernel=None, photmode=None,
         imgarr = img
 
     segm = detect_sources(imgarr, segment_threshold, npixels=source_box,
-                          filter_kernel=kernel)
+                          filter_kernel=kernel, connectivity=4)
+
+    log.debug("Creating segmentation map for {} ".format(outroot, kernel.shape))
+    if kernel is not None:
+        kernel_area = ((kernel.shape[0] // 2) ** 2) * np.pi
+        log.debug("   based on kernel shape of {}".format(kernel.shape))
+    else:
+        kernel_area = ((source_box // 2) ** 2) * np.pi
+        log.debug("   based on a default kernel.")
+
+    num_brightest = 10 if len(segm.areas) > 10 else len(segm.areas)
+    mean_area = np.mean(segm.areas)
+    max_area = np.sort(segm.areas)[-1 * num_brightest:].mean()
+    # This section looks for crowded fields where segments run into each other
+    # By reducing the size of the kernel used for segment detection, this can be minimized
+    # in crowded fields.  Also, mean area is used to try to avoid this logic for fields with
+    # several large extended sources in an otherwise empty field.
+    if max_area > 3422 and mean_area > (kernel_area / 2):  # largest > 33-pix radius source
```

Comment:
Define the magic number 3422.

Reply:
The magic number was moved to the top of the module with a slight re-definition and explanation in 219bfa6.